src/display/node_stream.js

```diff
@@ -26,29 +26,20 @@ if (typeof PDFJSDev !== "undefined" && PDFJSDev.test("MOZCENTRAL")) {
   );
 }
 
-const fileUriRegex = /^file:\/\/\/[a-zA-Z]:\//;
+const urlRegex = /^[a-z][a-z0-9\-+.]+:/i;
 
-function parseUrl(sourceUrl) {
-  const url = NodePackages.get("url");
-  const parsedUrl = url.parse(sourceUrl);
-  if (parsedUrl.protocol === "file:" || parsedUrl.host) {
-    return parsedUrl;
-  }
-  // Prepending 'file:///' to Windows absolute path.
-  if (/^[a-z]:[/\\]/i.test(sourceUrl)) {
-    return url.parse(`file:///${sourceUrl}`);
+function parseUrlOrPath(sourceUrl) {
```

Comment:
This is not 100% identical, but it only differs for a few edgecases on Windows:
- for `\a` the old one returned `file:///a` while the new one returns `file:///C:/a`
- for `\\?\a` the old one returned `file:////?\a` while the new one returns `file:///a`
- for one-char protocols followed by a hostname (`c://foo/bar`) the old one returned `c://foo/bar` while the new one returns `file:///c:/foo/bar`

These changes seem ok to me